### ISSUE TITLE
conda remove fails if package is disallowed

### ISSUE THREAD

**Issue by jakirkham** [conda/conda#2044]
If I have the following in my `.condarc`...

```
$ cat ~/.condarc
disallow:
  - mkl
```

...I am unable to remove the package if it was already installed.

```
$ conda remove mkl
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ..........
Error: specification 'mkl' is disallowed
```

**Comment by kalefranz** [comment]
You might be able to now.  I'm guessing disallow is completely broken at this point, because I can't think of any tests for it right now.

**Comment by kalefranz** [comment]
Disallow was completely broken in 4.4.  But should be fixed correctly in 4.5 with https://github.com/conda/conda/pull/6932
